edweiss-app/app/(app)/quiz/createQuizPage.tsx

```diff
@@ -0,0 +1,612 @@
+import RouteHeader from '@/components/core/header/RouteHeader';
+import ReactComponent, { ApplicationRoute } from '@/constants/Component';
+
+import TScrollView from '@/components/core/containers/TScrollView';
+import TTouchableOpacity from '@/components/core/containers/TTouchableOpacity';
+import TView from '@/components/core/containers/TView';
+import For from '@/components/core/For';
+import Icon from '@/components/core/Icon';
+import ModalContainer, { ModalContainerScrollView } from '@/components/core/modal/ModalContainer';
+import TText from '@/components/core/TText';
+import FancyButton from '@/components/input/FancyButton';
+import FancyTextInput from '@/components/input/FancyTextInput';
+import { MCQResultDisplay, TFResultDisplay } from '@/components/quiz/QuizComponents';
+import { callFunction, Collections, getDocument } from '@/config/firebase';
+import { useAuth } from '@/contexts/auth';
+import Quizzes, { QuizzesAttempts } from '@/model/quizzes';
+import { Time } from '@/utils/time';
+import { BottomSheetModal } from '@gorhom/bottom-sheet';
+import { BottomSheetModalMethods } from '@gorhom/bottom-sheet/lib/typescript/types';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import { router, useLocalSearchParams } from 'expo-router';
+import { t } from 'i18next';
+import React, { RefObject, useEffect, useRef, useState } from 'react';
+import Toast from 'react-native-toast-message';
+
+
+const CreateQuizPage: ApplicationRoute = () => {
+
+	const { courseId } = useLocalSearchParams()
+	const { uid } = useAuth()
+	const [exercises, setExercises] = useState<Quizzes.Exercise[]>([]);
+	const [quizName, setQuizName] = useState<string>("");
+	const [dueDate, setDueDate] = useState<Date | undefined>(undefined);
+	const [showDate, setShowDate] = useState<boolean>(false);
+	const [showTime, setShowTime] = useState<boolean>(false);
+
+
+	const addModalRef = useRef<BottomSheetModal>(null);
+
+	const publishQuizModalRef = useRef<BottomSheetModal>(null);
+
+
+	useEffect(() => {
+
+		console.log(JSON.stringify(exercises))
+	}, [exercises]);
+
+	async function checkUserType() {
+		const thisUser = await getDocument(Collections.users, uid);
+
+		if (thisUser?.data.type != "professor") {
+			return <TText> You are not authorized to create a quiz. </TText>;
+		}
+	}
+	checkUserType()
+	console.log("course id : " + courseId)
+
+
+	const onChangeDate = (selectedDate: Date | undefined) => {
+		if (selectedDate) {
+			setDueDate(selectedDate);
+			setShowDate(false);
+		}
+	};
+
+	const onChangeTime = (selectedDate: Date | undefined) => {
+		if (selectedDate) {
+			setDueDate(selectedDate);
+			setShowTime(false);
+		}
+	};
```

Comment:
This two functions are the same, you can create only one maybe. By exemple:

```
const handleDateTimeChange = (type: "date" | "time", selectedDate?: Date) => {
    if (selectedDate) {
        setDueDate(selectedDate);
        type === "date" ? setShowDate(false) : setShowTime(false);
    }
};
 
```

Reply:
Great idea! I'll do that for next PR.